waku-bindings/src/node/legacyfilter.rs

```diff
@@ -0,0 +1,89 @@
+//! Waku [filter](https://rfc.vac.dev/spec/36/#waku-filter) protocol related methods
+
+// std
+use std::ffi::CString;
+use std::time::Duration;
+// crates
+use libc::*;
+// internal
+use crate::general::Result;
+use crate::general::{LegacyFilterSubscription, PeerId};
+use crate::utils::{get_trampoline, handle_no_response};
+
+/// Creates a subscription in a lightnode for messages that matches a content filter and optionally a [`WakuPubSubTopic`](`crate::general::WakuPubSubTopic`)
+/// As per the [specification](https://rfc.vac.dev/spec/36/#extern-char-waku_legacy_filter_subscribechar-filterjson-char-peerid-int-timeoutms)
+pub fn waku_legacy_filter_subscribe(
+    filter_subscription: &LegacyFilterSubscription,
+    peer_id: PeerId,
+    timeout: Duration,
+) -> Result<()> {
+    let filter_subscription_ptr = CString::new(
+        serde_json::to_string(filter_subscription)
+            .expect("FilterSubscription should always succeed to serialize"),
+    )
+    .expect("FilterSubscription should always be able to be serialized")
+    .into_raw();
+    let peer_id_ptr = CString::new(peer_id)
+        .expect("PeerId should always be able to be serialized")
+        .into_raw();
```

Comment:
Seams like `peer_id` could benefit from an implementation of `Into<CString>` or `TryInto<CString>`

Same with other struct that get changed into `CString`